containers/src/Kokkos_DualView.hpp

```diff
@@ -361,10 +356,52 @@ class DualView : public ViewTraits<DataType, Properties...> {
       Kokkos::Impl::throw_runtime_exception(
           "DualView constructed with incompatible views");
     }
-    if (impl_dualview_stores_single_view && (d_view.data() != h_view.data()))
-      Kokkos::abort(
-          "DualView storing one View constructed from two different Views");
```

Comment:
We should keep this replacing `impl_dualview_stores_single_view` with `impl_dualview_is_single_device`. It's unrelated to any fencing issues in `Tpetra`.